d3ploy/demand_driven_deployment_inst.py

```diff
@@ -194,22 +201,37 @@ def enter_notify(self):
                 self.facility_constraintcommod,
                 self.facility_constraintval)
             commod_list = list(self.commodity_dict.keys())
+            for commod in commod_list:
+                self.installed_capacity[commod] = defaultdict(float)
+                self.installed_capacity[commod][0] = 0.
             for key, val in self.commodity_dict.items():
                 for key2, val2 in val.items():
                     if val2['constraint_commod'] != '0':
                         commod_list.append(val2['constraint_commod'])
-            commod_list = list(set(commod_list))
+            self.commod_list = list(set(commod_list))
             self.buffer_dict = di.build_buffer_dict(self.supply_buffer,
-                                                    commod_list)
+                                                    self.commod_list)
             self.buffer_type_dict = di.build_buffer_type_dict(
                 self.buffer_type, commod_list)
-            for commod in commod_list:
+            for commod in self.commod_list:
                 lib.TIME_SERIES_LISTENERS["supply" +
                                           commod].append(self.extract_supply)
                 lib.TIME_SERIES_LISTENERS["demand" +
                                           commod].append(self.extract_demand)
                 self.commodity_supply[commod] = defaultdict(float)
                 self.commodity_demand[commod] = defaultdict(float)
+            for child in self.children:
+                count = 0
+                for key, val in self.commodity_dict.items():
+                    for key2, val2 in val.items():
+                        if key2 == child.prototype:
+                            count = 1
+                            itscommod = key
+                if count == 0:
+                    raise Exception(
+                        'The {} facility that was added to the initial facility list must be included in facility_commod and facility_pref'.format(
+                            child.prototype))
+                self.installed_capacity[itscommod][0] = self.commodity_dict[itscommod][child.prototype]['cap']
```

Comment:
Shouldn't we be adding to, rather than replacing this value? What if there are many prototypes of this type?
Also, why access the zeroeth element explicitly rather than using that element as a key??

Why isn't this the strategy:
```suggestion
                self.installed_capacity[itscommod][child.prototype] += self.commodity_dict[itscommod][child.prototype]['cap']
```

Reply:
The `self.installed_capacity[itscommod][0]` refers to the installed capacity for that commodity at time 0.

Reply:
I edited the name of the key and vals in the loop so that is more clear. The point of doing that loop right before is to find out what commodity that `child` provides.